firestore/integration_test_internal/src/source_test.cc

```diff
@@ -0,0 +1,523 @@
+#include <future>
+
+#include "firebase/firestore.h"
+#include "firestore_integration_test.h"
+
+// These test cases are in sync with native iOS client SDK test
+//   Firestore/Example/Tests/Integration/API/FIRFirestoreSourceTests.mm
+// and native Android client SDK test
+//   firebase-firestore/src/androidTest/java/com/google/firebase/firestore/SourceTest.java
+
+namespace firebase {
+namespace firestore {
+
+using SourceTest = FirestoreIntegrationTest;
+
+TEST_F(SourceTest, TestGetDocumentWhileOnlineWithDefaultGetOptions) {
+  MapFieldValue initialData = {{"key", FieldValue::String("value")}};
+  DocumentReference docRef = DocumentWithData(initialData);
+
+  Future<DocumentSnapshot> future = docRef.Get();
+  WaitFor(future);
+
+  DocumentSnapshot snapshot = *future.result();
+  ASSERT_TRUE(snapshot.exists());
+  ASSERT_FALSE(snapshot.metadata().is_from_cache());
+  ASSERT_FALSE(snapshot.metadata().has_pending_writes());
+  ASSERT_EQ(initialData, snapshot.GetData());
+}
+
+TEST_F(SourceTest, TestGetCollectionWhileOnlineWithDefaultGetOptions) {
+  const std::map<std::string, MapFieldValue> initialDocs = {
+      {"doc1", {{"key1", FieldValue::String("value1")}}},
+      {"doc2", {{"key2", FieldValue::String("value2")}}},
+      {"doc3", {{"key3", FieldValue::String("value3")}}}};
+  CollectionReference colRef = Collection(initialDocs);
+
+  Future<QuerySnapshot> future = colRef.Get();
+  WaitFor(future);
+
+  QuerySnapshot querySnapshot = *future.result();
```

Comment:
Nit: it seems a little inconsistent that tests combine heavily abbreviated variable names (e.g. `col_ref`) with fully spelled out names (e.g. `query_snapshot`). Android uses abbreviated names throughout. I don't mind either, but perhaps choose a consistent approach? If you feel differently, let's discuss.